test/acceptance/features/environment.py

```diff
@@ -55,3 +56,13 @@ def before_scenario(_context, _scenario):
     _context.bindings = dict()
     output, code = cmd.run(f'{ctx.cli} get ns default -o jsonpath="{{.metadata.name}}"')
     assert code == 0, f"Checking connection to OS cluster by getting the 'default' project failed: {output}"
+
+
+def after_scenario(context, scenario):
+    delete_namespace=True
+    delete_namespace=os.getenv("DELETE_NAMESPACE")
+    namespace = os.getenv("TEST_NAMESPACE")
+    if (scenario.status == Status.passed) and (delete_namespace):
```

Comment:
Please consider a situation when the current scenarios passed, but one of the previously executed scenarios failed. In such case we don't want to delete the whole namespace because we would loose the resources for investigating or debugging the failed scenario.

The framework needs to remember that any scenario executed before the current one failed along with checking the current scenario's status and base the decision about the cleaning on that.